modules/structs/commit_status.go

```diff
@@ -18,14 +18,25 @@ const (
 	CommitStatusFailure CommitStatusState = "failure"
 )
 
+var commitStatusPriorities = map[CommitStatusState]int{
+	CommitStatusError:   0,
+	CommitStatusFailure: 1,
+	CommitStatusPending: 2,
+	CommitStatusSuccess: 3,
+}
+
 // NoBetterThan returns true if this State is no better than the given State
+// the function only handles the 4 states above
 func (css CommitStatusState) NoBetterThan(css2 CommitStatusState) bool {
-	commitStatusPriorities := map[CommitStatusState]int{
-		CommitStatusError:   0,
-		CommitStatusFailure: 1,
-		CommitStatusPending: 2,
-		CommitStatusSuccess: 3,
+	// NoBetterThan only handles the 4 states above
+	if _, exist := commitStatusPriorities[css]; !exist {
+		return false
```

Comment:
IMO, if you return false, this will not change anything. 
If css or css2 is empty, it is an invalid value and you should not call this compare function as it is not comparable.
If the caller doesn't notice this, maybe we will make the same mistake in the future.

So is it possible that add a function `IsValid` to check whether they are valid values?

Reply:
I don't think it's necessary to add another function, the above changes are enough